workspaces/quay/plugins/quay-backend/src/types.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Tag {
+  name: string;
+  isManifestList: boolean;
+  lastModified: string;
+  manifestDigest: string;
+  reversion: boolean;
+  size: number;
+  startTs?: number;
+  endTs?: number;
+  manifestList?: ManifestList;
+  expiration?: string;
+}
+
+export interface TagsResponse {
+  page: number;
+  hasAdditional: boolean;
+  tags: Tag[];
+}
+
+export interface Label {
+  id: string;
+  key: string;
+  value: string;
+  sourceType: string;
+  mediaType: string;
+}
+
+export interface LabelsResponse {
+  labels: Label[];
+}
+
+export interface Platform {
+  architecture: string;
+  os: string;
+  features?: string[];
+  variant?: string;
+  osVersion?: string;
+}
+
+export interface VulnerabilityMetadata {
+  updateBy: string;
+  repoName: string | null;
+  repoLink: string | null;
+  distroName: string;
+  distroVersion: string;
+  nvd: {
+    cvssV3: {
+      vectors: string;
+      score: number | string;
+    };
+  };
+}
+
+export enum VulnerabilitySeverity {
+  Critical = 'Critical',
+  High = 'High',
+  Medium = 'Medium',
+  Low = 'Low',
+  Negligible = 'Negligible',
+  None = 'None',
+  Unknown = 'Unknown',
+}
```

Comment:
I've been reading up on why enums are bad in typescript. Here are a few reference articles

* [why they are bad](https://www.totaltypescript.com/why-i-dont-like-typescript-enums)
* [alternative implementations](https://www.totaltypescript.com/books/total-typescript-essentials/deriving-types#using-as-const-for-javascript-style-enums)

You don't have to change this, but I thought I'd at least share this.

Reply:
Interesting, I'm going to save those links and consider that moving forward, thanks! I'll keep it as is for now, but going forward on another code base I'll consider the POJO approach.